packages/block-editor/src/components/keyboard-shortcuts/index.js

```diff
@@ -12,11 +12,14 @@ import { __ } from '@wordpress/i18n';
 
 function KeyboardShortcuts() {
 	// Shortcuts Logic
-	const { clientIds, rootBlocksClientIds } = useSelect( ( select ) => {
+	const { clientIds, rootBlocksClientIds, isCodeEditingEnabled } = useSelect( ( select ) => {
 		const { getSelectedBlockClientIds, getBlockOrder } = select( 'core/block-editor' );
+		const { codeEditingEnabled } = select( 'core/editor' ).getEditorSettings();
```

Comment:
We shouldn't be using the `core/editor` store in the block-editor store. Is  this config available in `select( 'core/block-editor' ).getSettings()`? Can we move the check to the action?

Reply:
@youknowriad thanks for your review, I find that `select( 'core/block-editor' ).getSettings()` does not have required `codeEditingEnabled` attribute.

Reply:
Does that mean we never checked for this option to show the block HTML mode? (without the shortcut)

Reply:
Excellent question. From my experience `getSettings( state ).codeEditingEnabled` evaluates to `undefined` when used in selector [canToggleBlockMode](https://github.com/WordPress/gutenberg/blob/10d6b0d36d2ae097400883f5ec4b859fe2072af8/packages/block-editor/src/store/selectors.js#L957). But it evaluates to `true` when used in `BlockModeToggle->compose->withSelect`.

Reply:
you're not using the same store `core/block-editor`, it's probably what's causing the difference. Again `core/editor` shouldn't be used in the `block-editor` package.

Reply:
does block-editor/store/selector not use core/block-editor?
I understand `core/editor` shouldn't be used, I tried that approach [here](https://github.com/dpkm95/gutenberg/tree/update/block-editor-toggle-block-mode-2) (another branch) but as I said `core/block-editor` state is not updated with codeEditingEnabled value as seen in the selector code.

Reply:
>  I said core/block-editor state is not updated with codeEditingEnabled value as seen in the selector code isCodeEditingEnabled evaluates toundefined.

This should be fixed.  and block-editor used everywhere. I can't say why the value is not propagating thought since it's already whitelisted here https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/provider/index.js#L91

Reply:
thanks for your input, will debug and fix it.